crates/deltalake-core/src/table/state_arrow.rs

```diff
@@ -167,13 +167,27 @@ impl DeltaTableState {
             })
             .collect::<HashMap<&str, _>>();
 
+        let physical_name_to_logical_name = metadata
+            .partition_columns
+            .iter()
+            .map(|name| {
+                let physical_name = metadata
+                    .schema
+                    .field_with_name(name)
+                    .unwrap()
```

Comment:
maybe add a quick comment why it is ok to unwrap here.

Reply:
Added that comment